From 30398197070a4eeefb214499c274a2c20123d516 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Tue, 11 Sep 2018 15:02:57 +0200
Subject: [PATCH 2/2] Removed kthread_run ad-hoc code.

This now builds upon kthread_create() and wake_up_process() just
like the Linux kernel does (sorry, no macro, since Windows doesn't
know how to handle blocks in macros).
---
 drbd/drbd_nl.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/drbd/drbd_nl.c b/drbd/drbd_nl.c
index 0b01db0..a28b2a9 100644
--- a/drbd/drbd_nl.c
+++ b/drbd/drbd_nl.c
@@ -928,22 +928,6 @@ static int _try_outdate_peer_async(void *data)
 	return 0;
 }
 
-struct task_struct * kthread_run(int (*threadfn)(void*), void *data, char *namefmt, ...)
-{
-    HANDLE hThread;
-    NTSTATUS status;
-    static struct task_struct ts = { 0 };
-
-    status = PsCreateSystemThread(&hThread, THREAD_ALL_ACCESS, NULL, NULL, NULL, threadfn, data);
-    if (status == STATUS_SUCCESS) {
-	ZwClose(hThread);
-	/* No data needed, only used to check for an error in conn_try_outdate_peer_async() */
-	return &ts;
-    } else {
-	return ERR_PTR(-ENOMEM);
-    }
-}
-
 void conn_try_outdate_peer_async(struct drbd_connection *connection)
 {
 	struct task_struct *opa;
-- 
2.17.0

